tools/cell_census_builder/tests/test_builder.py

```diff
@@ -82,6 +82,8 @@ def test_base_builder_creation(
             # Presence matrix should exist with the correct dimensions
             for exp_name in ["homo_sapiens", "mus_musculus"]:
                 fdpm = census[CENSUS_DATA_NAME][exp_name].ms[MEASUREMENT_RNA_NAME][FEATURE_DATASET_PRESENCE_MATRIX_NAME]
+                fdpm_matrix = fdpm.read().coos().concat()
+                assert fdpm_matrix.shape[0] == 4  # 4 datasets
```

Comment:
Each presence matrix should only have rows for the datasets that the experiment it is associated with. So 2 (distinct) datasets in each presence matrix for this test, 4 features each, 8 rows total.  See [discussion](https://czi-sci.slack.com/archives/C048BN7K6U8/p1676562041122019).  Let me know if I'm misunderstanding.

Reply:
Yeah, this assert is wrong.  Andrew has it right as I read the conftest code.

Remember, per Pablo's spec, it is the number of datasets that contribute data to the _specific_ experiment, not the number in the total census (i.e., union of all organisms)

Reply:
As mentioned in the (edited) description, this wasn't the previous behavior which is what this PR is trying to restore. Since that is not compliant with the SOMA abstract spec, we'll probably want to close this and work towards a more permanent fix.

Reply:
Yeah, I was misunderstanding the issue at hand. 

Here are some suggested constraints to assert - ie. what we would expect to be true:

1. For all experiment's presence matrix, all dimension 0 coordinates (aka the dataset DataFrame's soma_joinid) should be > 0 and <= the max soma_joinid in the datasets table.
2. for all experiment's presence matrix, the shape[0] == max(datasets soma_joinid) + 1 

Important assumption above:  the rows in the datasets DataFrame are assigned 0..n_datasets.  This is just a convention (they could be anything, but we choose to sequentially assign them starting at zero).

So I would change your assertion to read:

```
assert fdpm_matrix.shape[0] == max(returned_datasets.soma_joinid) + 1
```

Reply:
BTW, to clear up the above comment - the spec is _silent_ on the shape of the presence matrix, and only specifies its contents.  I believe that fixing the shape as proposed by this PR is consistent with the spec.